ballista/core/proto/ballista.proto

```diff
@@ -921,6 +921,24 @@ message RunningTaskInfo {
   uint32 stage_id = 3;
 }
 
+message ShortCircuitRegisterRequest {
+  string task_identity = 1;
+  optional uint64 row_count_limit = 2;
+  optional uint64 byte_count_limit = 3;
+}
```

Comment:
Why do we need to register? Shouldn't the scheduler handle this internally during planning and the executors just send updates?

Reply:
That way we wouldn't need to handle the lifecycle at all on the executor

Reply:
This approach is problematic, as you would have to introspect the new execution plan to try and find all nodes that are going to be using this functionality, which is not possible in a generic manner. So, it would have to be implemented on DQE side. So I think it's a fair tradeoff to simply make one initial request to register at the beginning of the stream, as that channel of communication already has to exist regardless.

And actually I do want to manage the lifecycle so I can accumulate updates on the executor and limit how often a task sends an update, so the scheduler does not get bombarded.

On that note, I did just realized that I forget to delete the map entries in the client so I will fix that.

Reply:
> I can accumulate updates on the executor and limit how often a task sends an update, so the scheduler does not get bombarded.

This is a very good idea but I don't think it requires that we manage the lifecycle on the executor. Just that we do batching on the executor side. 

I just think that the scheduler is already managing the job lifecycle (and already introspects the plan multiple times) so it seems like we can handle it quite easily without adding additional RPC traffic. I think we would also be required to manage the lifecycle to some extent on the scheduler since the executor only knows about it's tasks so the accumulator for the stage would need to be de-registered by the scheduler anyway.

Reply:
Fixed